src/compiler/checker.ts

```diff
@@ -12701,6 +12701,19 @@ namespace ts {
                     return anyType;
                 }
                 if (accessExpression && !isConstEnumObjectType(objectType)) {
+                    if (isObjectLiteralType(objectType)) {
+                        if (indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {

```

Comment:
Hm, when you index with a literal type, we should probably keep the "Property does not exist on type" error. Since doing something like `{field: 1, another: 2}["feild"]` is almost definitely an error.

Reply:
@RyanCavanaugh you made the original suggestion, you have an opinion?